gm4_damaging_stonecutters/data/damaging_stonecutters/functions/pulse_check.mcfunction

```diff
@@ -0,0 +1,4 @@
+#compares assigned run tick to current tick and calls main
+#execute if score current_tick gm4_clock_tick = damaging_stonecutters gm4_clock_tick run function damaging_stonecutters:main
+
+execute as @e[type=!#damaging_stonecutters:no_target] at @s run function damaging_stonecutters:check_block
```

Comment:
You're only damaging entities every 10 ticks anyways, so a slower clock should be used (perhaps every 8 ticks). This will not only make the module more efficient, but will also simplify the code a lot while getting rid of the scoreboard.

Reply:
I used a separate clock so that it damaged you the second you went onto the stonecutter - it felt better without a delay